src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -326,3 +326,9 @@ def _update(info, lun, value):
             except ValueError:
                 raise CLIError("A sku ID is incorrect.\n{}".format(usage_msg))
             _update(info_dict, lun, value)
+
+
+# deprecate --name -n in favor of --disk
```

Comment:
I understand. But because [`az vmss disk attach`](https://docs.microsoft.com/en-us/cli/azure/vmss/disk?view=azure-cli-latest#az-vmss-disk-attach) already uses `--disk` and `--name` for different parameters, If i deprecate `--disk` for `--name` in az vm disk and az vm unmanaged-disk then I would have to either make a breaking change in `az vmss disk` (unable to deprecate two parameters where one option is deprecated for another deprecated option) or make it inconsistent with `az vm disk` and `az vm unmanaged-disk`. 

I would use `--disk-name` instead, if we insist on changing --disk, however `az vm attach --disk` accepts a name or id parameter.